python/lsst/daf/persistence/repository.py

```diff
@@ -233,15 +235,20 @@ def _initMapper(self, repoCfg):
                 mapperArgs = copy.copy(repoCfg['mapperArgs'])
                 if mapperArgs is None:
                     mapperArgs = {}
+                # so that root doesn't have to be redundantly passed in cfgs, if root is specified in the
+                # storage and if it is an argument to the mapper, make sure that it's present in mapperArgs.
                 if 'root' not in mapperArgs:
-                    mapperArgs['root'] = self._access.root()
+                    for c in inspect.getmro(mapper):
```

Comment:
This is really pretty ugly.  I think you would be much better off defining some class attributes or methods on `Mapper` that derived classes could use to indicate that they need certain arguments.  Yes, this means changing existing `Mapper` subclasses, but I think all of them are in tests or are `CameraMapper` subclasses.